appscale/tools/appscale_tools.py

```diff
@@ -505,14 +505,26 @@ def run_instances(cls, options):
       AppScaleLogger.log("\nPlease wait for AppScale to prepare your machines "
                          "for use. This can take few minutes.")
 
+      head_node.public_ip = public_ips[0]
```

Comment:
Is this change needed because only the first node gets a public IP address in Azure? Since the `start_all_nodes` function does not imply this, I think an explanatory comment would be helpful here.